survey_admin/templates/show_surveys.html

```diff
@@ -19,6 +19,8 @@
 		<tr>
 			<td><a href="{{ url_for('survey_page', surveyid=s.survey_info_id) }}">{{ s.survey_name }}</a></td>
 			<td>{{ s.description }}</td>
+
+            <td><a href="{{ url_for('edit_survey_form', survey_info_id=s.survey_info_id) }}" class="btn btn-warning btn-sm">Edit</a></td>
```

Comment:
The id isn't displayed. Let me explain. 

1. some user goes to `localhost:8989/showsurveys` because on the top bar they clicked `Surveys` then `View Surveys`.
2. This page (using the `show_surveys.html` template) appears which looks like this in the browser:

![screenshot from 2018-09-07 20-42-41](https://user-images.githubusercontent.com/19616012/45251249-9eb5d280-b2de-11e8-8db1-9410566a8e35.png)

3. The href (not visible to the naked eye unless inspecting the html) is dynamically created for each survey as well as for the button to the right of the survey. Using `url_for` in the line you're looking at, I'm telling Flask to create the href such that it will go to this route in `views.py`:

```python
@app.route('/editsurveyform/<int:survey_info_id>', methods=['GET', 'POST'])
@login_required
def edit_survey_form(survey_info_id):
    """Route used to display the edit_survey_form.html page

    GET: renders the edit_survey_form page for the given survey_info_id
    POST: submits desired changes to the database, then redirects user to show_surveys
    """


    form = EditSurveyForm()
    form.survey_questions.choices = [(q.question_id, "(" + str(q.question_id) + ") " + q.question_text) for q in Question.query.order_by("question_id")]
    currently_selected_questions = [q.question_id for q in SurveyQuestion.query.filter_by(survey_info_id=survey_info_id)]


    if request.method == 'GET':
        return render_template('edit_survey_form.html', form=form, survey_info_id=survey_info_id, currently_selected_questions=",".join(map(str, currently_selected_questions)))
    elif request.method == 'POST':


        if form.validate_on_submit():
            newly_selected_questions = []


            for question in form.survey_questions:
                if question.checked:
                    newly_selected_questions.append(question.data)


            # delete the old
            print("Remove previously selected questions with id: " + ', '.join(map(str, currently_selected_questions)))
            db_session.query(SurveyQuestion).filter(SurveyQuestion.survey_info_id==survey_info_id).delete()


            # insert the new
            for question_id in newly_selected_questions:
                db_session.add(SurveyQuestion(survey_info_id, question_id, 0))


            db_session.commit()


            print("previous questions in this survey")
            print(currently_selected_questions)


            print("newly selected questions in this survey")
            print(newly_selected_questions)


            return redirect(url_for('show_surveys'))
        else:
            return render_template('edit_survey_form.html', form=form, survey_info_id=survey_info_id,
                                   currently_selected_questions=",".join(map(str, currently_selected_questions)))


```
Notice in `app.route` there is `<int:survey_info_id>'`. Also `survey_info_id` is an argument to the function right below. `survey_info_id` is required because that will get passed down into the function so that I know to generate the `edit_survey_form` page based on a specific `survey_info_id`. 

4. The page that then gets displayed to the user is a template. And so we need to know what information to fill it with. Based on the `survey_info_id` passed, we know what content to fill the page with.

So this part really has nothing to do with the id's being serial (or auto-increment or whatever we call it). What is displayed to the user based on this line in question is only a yellow button with the word "Edit".